euphonic/spectra.py

```diff
@@ -473,9 +527,24 @@ def _type_check(spectrum):
             assert spectrum.x_tick_labels == x_tick_labels
             y_data_magnitude[i + 1, :] = spectrum.y_data.magnitude
 
-        y_data = Quantity(y_data_magnitude, y_data_units)
+        # Process metadata
+        metadata = deepcopy(spectra[0]).metadata
```

Comment:
What happens if spectra[1] has totally different metadata?

I think ideally, with the exception of `labels`, we would retain metadata that is the same for all component spectra and either discard or reorganise data that differs by line.

Maybe `labels` shouldn't be a special case, and there should be a general pattern for Spectrum1dCollection metadata such that

```python
spec1.metadata = {'element': 'K', 'label': 'Potassium 1', 'XC': 'PBE', 'calctime': 107.43}
spec2.metadata = {'element': 'K', 'label': 'Potassium 2', 'XC': 'PBE', 'calctime': 105.23}

Spectrum1DCollection.from_spectra([spec1, spec2]).metadata = {
    'element': 'K',
    'XC': 'PBE',
    'line_data': [{'label': 'Potassium 1', 'calctime': 107.43},
                  {'label': 'Potassium 2', 'calctime': 105.23}]}
```

Pros:
- completely reversible to recover consistent Spectrum1D.metada when iterating
- `'label'` and `'labels'` are not inconsistent between metadata dictionaries, and are only used in function arguments as appropriate.
- extends naturally without adding more special cases, would play very nicely with a `plot_args` option for storing desired linewidths, colours etc.

Cons:
- I appreciate this may not be a welcome suggestion after all the work you did to implement my previous suggestion :disappointed:

Reply:
Yep, I agree this would probably be a better solution, these things aren't always obvious on the first pass. I'll have a go at changing the implementation and then address your other comments based on the state of the code once I'm finished..

Reply:
When I implemented something like this for ASE, I found `set(some_dict.items())` to be a useful tool: a set of 2-tuples allows you to easily find the common and different key-value pairs.